packages/compose/src/hooks/use-focus-on-mount/index.js

```diff
@@ -27,8 +27,16 @@ import { focus } from '@wordpress/dom';
  */
 export default function useFocusOnMount( focusOnMount ) {
 	const didMount = useRef( false );
+	const focusOnMountRef = useRef( focusOnMount );
+	useEffect( () => {
+		focusOnMountRef.current = focusOnMount;
+	}, [ focusOnMount ] );
 	return useCallback( ( node ) => {
-		if ( ! node || didMount.current === true ) {
+		if (
+			! node ||
+			didMount.current === true ||
+			focusOnMountRef.current === false
```

Comment:
To disable the behaviour, the implementor should just not include the ref?

Reply:
if you use useDialog, you need an option to do so, we can move the logic to there of course, but it seemed nicer from devx point of view to just use the existing config.